_includes/tutorials/message-ordering/kafka/code/src/main/java/io/confluent/developer/KafkaProducerApplication.java

```diff
@@ -0,0 +1,103 @@
+package io.confluent.developer;
+
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+public class KafkaProducerApplication {
+
+    private final Producer<String, String> producer;
+    final String outTopic;
+
+    public KafkaProducerApplication(final Producer<String, String> producer,
+                                    final String topic) {
+        this.producer = producer;
+        outTopic = topic;
+    }
+
+    public void produce(final String message) {
+        final String[] parts = message.split("-");
+        final String key, value;
+        if (parts.length > 1) {
+            key = parts[0];
+            value = parts[1];
+        } else {
+            key = null;
+            value = parts[0];
+        }
+        final ProducerRecord<String, String> producerRecord = new ProducerRecord<>(outTopic, key, value);
+        producer.send(producerRecord,
+            new Callback() {
```

Comment:
I'd probably write this as a lambda instead of an anonymous inner class (more concise, and implicitly gives you access to the `producerRecord` inside the callback), but not critical.

Reply:
@bbejeck @mikebin very cool suggestion @mikebin .  WDYT about the implementation https://github.com/confluentinc/kafka-tutorials/pull/587/commits/1d8572ba985096b03078355bb948487e8837bb47 ?

Reply:
LGTM!